grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/query/Query.java

```diff
@@ -334,9 +333,9 @@ public void unify(Map<String, String> unifiers) {
 
     public MatchQuery getMatchQuery() {
         if (selectVars.isEmpty())
-            return Queries.matchNoInfer(getPattern()).withGraph(graph);
+            return graph.graql().setInference(false).match(getPattern());
         else
-            return Queries.matchNoInfer(getPattern()).select(selectVars).withGraph(graph);
+            return graph.graql().setInference(false).match(getPattern()).select(selectVars);
     }
 
     public Map<String, Type> getVarTypeMap() {
```

Comment:
this shouldn't have any effect, the input match queries are only used to form the atoms on construction and later discarded